price-tolerance/src/main.nr

```diff
@@ -34,8 +49,16 @@ fn main(validators: pub [Float; MAX_VALIDATORS], tolerance: pub Float) {
             //  perc_chng = |values[i] - median| / median
             let perc_chng = div(abs(sub(values[i], median)), median);
             if lte(perc_chng, tolerance) {
+                correct_count += 1;
                 printlnF(values[i]);
             }
         }
     }
+
+    print("Number of correct validators: ");
+    print(correct_count);
+    print(" of ");
+    println(n);
+
+    (values, correct_count, n)
```

Comment:
Returning values directly means we also return the invalid ones. Though the consumer of the output may be able to get back the correct values (assuming he knows that we use the median value, and values are sorted), I don't believe this is good design.

We should at the very least return an array which has only the correct values and probably trivial values for padding.

Reply:
This is true, we should fill a new array with only the correct values where the threshold check is performed.